src/arkode/arkode_ls.c

```diff
@@ -114,7 +114,8 @@ int ARKodeSetLinearSolver(void* arkode_mem, SUNLinearSolver LS, SUNMatrix A)
   /* Check for compatible LS type, matrix and "atimes" support */
   if (iterative)
   {
-    if (ark_mem->tempv1->ops->nvgetlength == NULL)
+    if ((ark_mem->tempv1->ops->nvgetlength == NULL) ||
+        (ark_mem->tempv1->ops->nvdotprod == NULL))
```

Comment:
Move the `nvdotprod` check to `ARKodeSetLSNormFactor` and `ARKodeSetMassLSNormFactor` as it's only needed if the input to those functions is < 0.